booking/src/main/java/org/dcsa/conformance/standards/booking/model/PersistableCarrierBooking.java

```diff
@@ -487,6 +493,27 @@ private void addCharge(ObjectNode booking) {
     }
   }
 
+  private void addAdvancedManifestFilings(ObjectNode booking) {
+    ArrayNode advanceManifestFilings;
+    if (booking.get("advanceManifestFilings") instanceof ArrayNode advanceManifestFilingsNode) {
+      advanceManifestFilings = advanceManifestFilingsNode;
+    } else {
+      advanceManifestFilings = booking.putArray("advanceManifestFilings");
+    }
+    if (advanceManifestFilings.isEmpty()) {
+      advanceManifestFilings
+        .addObject()
+        .put("manifestTypeCode", "ACI")
+        .put("countryCode", "EG");
+    } else {
+      advanceManifestFilings
+        .addObject()
+        .put("manifestTypeCode", "ACE")
+        .put("countryCode", "US");
+    }
```

Comment:
This will still cause duplicates if the `addAdvancedManifestFilings` is called multiple for the same booking (we are now at thrice rather than twice, but the underlying bug remains).

Reply:
The example code would solve my concern indeed.